diff -Naur libgcrypt-1.7.2.tpg/random/random-csprng.c libgcrypt-1.7.2/random/random-csprng.c
--- libgcrypt-1.7.2.tpg/random/random-csprng.c	2016-04-07 15:30:08.000000000 +0000
+++ libgcrypt-1.7.2/random/random-csprng.c	2016-07-16 13:18:25.326594220 +0000
@@ -1108,8 +1108,7 @@
              enum random_origins, size_t, int);
 
 #if USE_RNDLINUX
-  if ( !access (NAME_OF_DEV_RANDOM, R_OK)
-       && !access (NAME_OF_DEV_URANDOM, R_OK))
+  if ( !access (NAME_OF_DEV_URANDOM, R_OK))
     {
       fnc = _gcry_rndlinux_gather_random;
       return fnc;
diff -Naur libgcrypt-1.7.2.tpg/random/rndlinux.c libgcrypt-1.7.2/random/rndlinux.c
--- libgcrypt-1.7.2.tpg/random/rndlinux.c	2016-07-16 13:13:45.616307308 +0000
+++ libgcrypt-1.7.2/random/rndlinux.c	2016-07-16 13:19:20.693010608 +0000
@@ -170,7 +170,11 @@
     {
       if (fd_random == -1)
         {
-          fd_random = open_device (NAME_OF_DEV_RANDOM, (ever_opened & 1));
+          /* We try to open /dev/random first but in case the open fails
+             we gracefully retry with /dev/urandom. */
+          fd_random = open_device (NAME_OF_DEV_RANDOM, 0, 0);
+          if (fd_random == -1)
+            fd_random = open_device (NAME_OF_DEV_URANDOM, (ever_opened & 1), 1);
           ever_opened |= 1;
         }
       fd = fd_random;
