diff -up libgcrypt-1.11.2/configure.ac.omv~ libgcrypt-1.11.2/configure.ac
--- libgcrypt-1.11.2/configure.ac.omv~	2025-09-24 01:00:15.243750313 +0200
+++ libgcrypt-1.11.2/configure.ac	2025-09-24 01:02:22.519406615 +0200
@@ -1472,14 +1472,17 @@ fi
 #
 # Check whether compiler support '__thread' storage class specifier.
 #
-AC_CACHE_CHECK([whether compiler supports '__thread' storage class specifier],
-       [gcry_cv_gcc_storage_class__thread],
-       [gcry_cv_gcc_storage_class__thread=no
-        AC_COMPILE_IFELSE([AC_LANG_PROGRAM(
-          [[static __thread int bar;]],
-          [[(void)bar;]]
-          )],
-          [gcry_cv_gcc_storage_class__thread=yes])])
+#AC_CACHE_CHECK([whether compiler supports '__thread' storage class specifier],
+#       [gcry_cv_gcc_storage_class__thread],
+#       [gcry_cv_gcc_storage_class__thread=no
+#        AC_COMPILE_IFELSE([AC_LANG_PROGRAM(
+#          [[static __thread int bar;]],
+#          [[(void)bar;]]
+#          )],
+#          [gcry_cv_gcc_storage_class__thread=yes])])
+# The test above has a bogus failure when using -fprofile-use= with clang 21.x
+# But we will never target anything too primitive to have __thread, so.....
+gcry_cv_gcc_storage_class__thread=yes
 if test "$gcry_cv_gcc_storage_class__thread" = "yes" ; then
    AC_DEFINE(HAVE_GCC_STORAGE_CLASS__THREAD,1,
      [Defined if compiler supports "__thread" storage class specifier])
