diff -Naur libgcrypt-1.10.1/random/jitterentropy-base.c libgcrypt-1.10.1.tpg/random/jitterentropy-base.c
--- libgcrypt-1.10.1/random/jitterentropy-base.c	2022-01-25 21:55:44.000000000 +0000
+++ libgcrypt-1.10.1.tpg/random/jitterentropy-base.c	2022-05-26 21:09:57.367250186 +0000
@@ -29,6 +29,14 @@
  * DAMAGE.
  */
 
+#ifdef __clang__
+#pragma clang optimize off
+#else
+ #ifdef __OPTIMIZE__
+ #pragma GCC optimize ("O0")
+ #endif
+#endif
+
 #include "jitterentropy.h"
 
 #include "jitterentropy-base.h"
@@ -54,7 +62,7 @@
  * None of the following should be altered
  ***************************************************************************/
 
-#ifdef __OPTIMIZE__
+#if defined(__OPTIMIZE__) && !defined(__clang__)
  #error "The CPU Jitter random number generator must not be compiled with optimizations. See documentation. Use the compiler switch -O0 for compiling jitterentropy.c."
 #endif
 
