diff -up libgcrypt-1.12.0/src/libgcrypt-config.in.1~ libgcrypt-1.12.0/src/libgcrypt-config.in
--- libgcrypt-1.12.0/src/libgcrypt-config.in.1~	2025-09-23 15:14:23.000000000 +0200
+++ libgcrypt-1.12.0/src/libgcrypt-config.in	2026-01-29 16:45:52.432865002 +0100
@@ -16,7 +16,6 @@ prefix="@prefix@"
 exec_prefix="@exec_prefix@"
 version="@PACKAGE_VERSION@"
 includedir="@includedir@"
-libdir="@libdir@"
 gpg_error_libs="@GPG_ERROR_LIBS@"
 gpg_error_cflags="@GPG_ERROR_CFLAGS@"
 
@@ -165,11 +164,6 @@ if test "$echo_libs" = "yes"; then
     libdirs=""
     libs_final="$libs"
 
-    # Set up `libdirs'.
-    if test "x$libdir" != "x/usr/lib" && test "x$libdir" != "x/lib"; then
-	libdirs="-L$libdir"
-    fi
-
     # Set up `libs_final'.
     libs_final="$libs_final $gpg_error_libs"
 
