diff -uraN libgcrypt-1.9.4/random/jitterentropy-base.c omv-libgcrypt-1.9.4/random/jitterentropy-base.c
--- libgcrypt-1.9.4/random/jitterentropy-base.c	2021-08-22 14:57:34.000000000 +0200
+++ omv-libgcrypt-1.9.4/random/jitterentropy-base.c	2021-08-26 12:47:59.639114492 +0200
@@ -50,15 +50,19 @@
  */
 
 #undef _FORTIFY_SOURCE
+#ifdef __clang__
+#pragma clang optimize off
+#else
 #ifdef __OPTIMIZE__
 #pragma GCC optimize ("O0")
 #endif
+#endif
 
 #include "jitterentropy.h"
 
 #ifndef CONFIG_CRYPTO_CPU_JITTERENTROPY_STAT
  /* only check optimization in a compilation for real work */
- #ifdef __OPTIMIZE__
+ #if defined(__OPTIMIZE__) && !defined(__clang__)
   #error "The CPU Jitter random number generator must not be compiled with optimizations. See documentation. Use the compiler switch -O0 for compiling jitterentropy-base.c."
  #endif
 #endif
